### PR TITLE
Feat: custom editable fields

### PR DESCRIPTION
**Pull request by josepfo**

### CODE REVIEW

**Comment by josepfo** [comment]
[(frontend) Add custom, editable fields to customer profile page](https://trello.com/c/ooj9IUZW/1015-frontend-add-custom-editable-fields-to-customer-profile-page)

File: new-lamassu-admin/src/pages/Customers/CustomerData.js [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+import styles from './CustomerData.styles.js'
+import { EditableCard } from './components'
+import { getName } from './helper.js'
+
+const useStyles = makeStyles(styles)
+
+const imageWidth = 165
+const imageHeight = 45
+const popupImageWidth = 360
+const popupImageHeight = 240
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/CustomerData.js]:
```suggestion
const IMAGE_WIDTH = 165
const IMAGE_HEIGHT = 45
const POPUP_IMAGE_WIDTH = 360
const POPUP_IMAGE_HEIGHT = 240
```

File: new-lamassu-admin/src/pages/Customers/CustomerData.js [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+  const sanctions = R.path(['sanctions'])(customer)
+  const sanctionsAt = R.path(['sanctionsAt'])(customer)
+  const sanctionsDisplay = !sanctionsAt
+    ? 'Not checked yet'
+    : sanctions
+    ? 'Passed'
+    : 'Failed'
+
+  const customEntries = null // get customer custom entries
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/CustomerData.js:78]: It's missing the database query here

File: new-lamassu-admin/src/pages/Customers/Wizard.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  const [selectedValues, setSelectedValues] = useState(null)
+  const [liveValues, setLiveValues] = useState({})
+
+  const [{ step, config }, setState] = useState({
+    step: 1
+  })
+
+  // TODO forms error handling
+  const isLastStep = step === LAST_STEP
+  const stepOptions = getStep(step, selectedValues, liveValues)
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/Wizard.js]: I'm not understanding the need for a component to keep the form values in it, can you explain?

File: new-lamassu-admin/src/pages/Customers/Wizard.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+              selectedValues={selectedValues}
+              {...stepOptions.props}
+            />
+            <div className={classes.submit}>
+              {error && <ErrorMessage>Failed to save</ErrorMessage>}
+              <Button className={classes.button} type="submit">
+                {isLastStep ? 'Add Data' : 'Next'}
+              </Button>
+            </div>
+          </Form>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/Wizard.js]: You can get `values`, `errors` and `touched` on a Formik form with this. Shouldn't be needed a component just for that

```suggestion
          {({ values, touched, errors }) => (<Form className={classes.form}>
            <GetValues setValues={setLiveValues} />
            <stepOptions.Component
              selectedValues={selectedValues}
              {...stepOptions.props}
            />
            <div className={classes.submit}>
              {error && <ErrorMessage>Failed to save</ErrorMessage>}
              <Button className={classes.button} type="submit">
                {isLastStep ? 'Add Data' : 'Next'}
              </Button>
            </div>
          </Form>)}
```

File: new-lamassu-admin/src/pages/Customers/components/CustomerSidebar.styles.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    cursor: 'pointer'
+  },
+  activeLink: {
+    display: 'flex',
+    alignItems: 'center',
+    extend: tl2,
+    color: white,
+    backgroundColor: offDarkColor,
+    '&:first-child': {
+      borderRadius: '5px 5px 0px 0px'
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/components/CustomerSidebar.styles.js]: I believe you can simplify with this

```suggestion
      borderRadius: [ 5, 5, 0, 0 ]
```

File: new-lamassu-admin/src/pages/Customers/components/CustomerSidebar.styles.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    '&:last-child': {
+      borderRadius: '0px 0px 5px 5px'
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/pages/Customers/components/CustomerSidebar.styles.js]: Same here

File: new-lamassu-admin/src/styling/icons/action/edit/comet.svg [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="22px" height="22px" viewBox="0 0 22 22" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <!-- Generator: Sketch 62 (91390) - https://sketch.com -->
+    <desc>Created with Sketch.</desc>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/action/edit/comet.svg:3]: You can remove this line I think, as `title` nodes show up on mouseover

File: new-lamassu-admin/src/styling/icons/button/data/white.svg [hunk 8]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/data/white</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/data/white.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/data/zodiac.svg [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/data/zodiac</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/data/zodiac.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/discount/comet.svg [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/loyalty/comet</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/discount/comet.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/discount/white.svg [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/loyalty/white</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/discount/white.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/discount/zodiac.svg [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/loyalty/zodiac</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/discount/zodiac.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/photo/white-resized.svg [hunk 13]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/photo/white-resized</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/photo/white-resized.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/photo/zodiac-resized.svg [hunk 14]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/photo/zodiac-resized</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/photo/zodiac-resized.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/replace/white.svg [hunk 15]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/replace/white</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/replace/white.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/replace/zodiac.svg [hunk 16]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/replace/zodiac</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/replace/zodiac.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/upload-file/white-resized.svg [hunk 17]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/upload-file/white-resized</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/upload-file/white-resized.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/upload-file/white.svg [hunk 18]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/upload-file/white</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/upload-file/white.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/upload-file/zodiac-resized.svg [hunk 19]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/upload-file/zodiac-resized</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/upload-file/zodiac-resized.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/button/upload-file/zodiac.svg [hunk 20]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="12px" height="12px" viewBox="0 0 12 12" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/button/upload-file/zodiac</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/button/upload-file/zodiac.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/circle buttons/customer-list-view/white.svg [hunk 21]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/sf-small/listing/white</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/circle buttons/customer-list-view/white.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/circle buttons/customer-list-view/zodiac.svg [hunk 22]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/sf-small/listing/zodiac</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/circle buttons/customer-list-view/zodiac.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/circle buttons/overview/comet.svg [hunk 23]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/sf-small/overview/comet</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/circle buttons/overview/comet.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/circle buttons/overview/white.svg [hunk 24]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/sf-small/overview/white</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/circle buttons/overview/white.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/circle buttons/overview/zodiac.svg [hunk 25]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/sf-small/overview/zodiac</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/circle buttons/overview/zodiac.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/customer-nav/data/comet.svg [hunk 26]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/customer-nav/data/comet</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/customer-nav/data/comet.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/customer-nav/data/white.svg [hunk 27]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/customer-nav/data/white</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/customer-nav/data/white.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/customer-nav/data/zodiac.svg [hunk 28]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/customer-nav/data/zodiac</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/customer-nav/data/zodiac.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/customer-nav/overview/comet.svg [hunk 29]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/customer-nav/overview/comet</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/customer-nav/overview/comet.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/customer-nav/overview/white.svg [hunk 30]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/customer-nav/overview/white</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/customer-nav/overview/white.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/customer-nav/overview/zodiac.svg [hunk 31]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/customer-nav/overview/zodiac</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/customer-nav/overview/zodiac.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/file/comet.svg [hunk 32]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="32px" height="44px" viewBox="0 0 32 44" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/file/zip/comet</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/file/comet.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/file/spring.svg [hunk 33]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="32px" height="44px" viewBox="0 0 32 44" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/file/zip/spring</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/file/spring.svg]: Here as well

File: new-lamassu-admin/src/styling/icons/file/tomato.svg [hunk 34]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="32px" height="44px" viewBox="0 0 32 44" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>icon/file/zip/tomato</title>
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/styling/icons/file/tomato.svg]: Here as well

**Review by chaotixkilla — CHANGES_REQUESTED** [review]

File: new-lamassu-admin/src/pages/Customers/CustomerData.js (hunk 2 above)
Comment (josepfo) [reply new-lamassu-admin/src/pages/Customers/CustomerData.js:78]: it's not merged yet

File: new-lamassu-admin/src/pages/Customers/Wizard.js (hunk 3 above)
Comment (josepfo) [reply new-lamassu-admin/src/pages/Customers/Wizard.js]: Yeah, not needed here at all. It's some leftovers from another Wizard

**Review by chaotixkilla — APPROVED** [review]

**Pull request merged**